teamcity/datadog_checks/teamcity/check.py

```diff
@@ -0,0 +1,43 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from copy import deepcopy
+from urllib.parse import urlparse
+
+from datadog_checks.base import OpenMetricsBaseCheckV2, is_affirmative
+
+from .metrics import METRIC_MAP
+
+
+class TeamCityCheckV2(OpenMetricsBaseCheckV2):
+    __NAMESPACE__ = 'teamcity'
+    DEFAULT_METRIC_LIMIT = 0
+
+    DEFAULT_METRICS_URL = "/{}/app/metrics"
+    EXPERIMENTAL_METRICS_URL = "/{}/app/metrics?experimental=true"
+
+    def __init__(self, name, init_config, instances):
+        super(TeamCityCheckV2, self).__init__(name, init_config, instances)
+        self.basic_http_auth = is_affirmative(
+            self.instance.get('basic_http_authentication', bool(self.instance.get('password', False)))
+        )
+        self.auth_type = 'httpAuth' if self.basic_http_auth else 'guestAuth'
+        parsed_endpoint = urlparse(self.instance.get('server'))
+        self.server_url = "{}://{}".format(parsed_endpoint.scheme, parsed_endpoint.netloc)
+        self.metrics_endpoint = ''
+
+        experimental_metrics = is_affirmative(self.instance.get('experimental_metrics', False))
+
+        if experimental_metrics:
+            self.metrics_endpoint = self.EXPERIMENTAL_METRICS_URL.format(self.auth_type)
+        else:
+            self.metrics_endpoint = self.DEFAULT_METRICS_URL.format(self.auth_type)
```

Comment:
Does the experimental endpoint include all the non experimental ones or should the user define two instances?